src/metabase/upload.clj

```diff
@@ -176,10 +193,12 @@
 ;;;; +------------------+
 
 (defn unique-table-name
-  "Append the current datetime to the given name to create a unique table name."
-  [table-name]
-  (str (u/slugify table-name)
-       (t/format "_yyyyMMddHHmmss" (t/local-date-time))))
+  "Append the current datetime to the given name to create a unique table name. The resulting name will be short enough for the given driver (truncating the supplised `table-name` if necessary)."
+  [driver table-name]
+  (let [time-format "_yyyyMMddHHmmss"]
+    (str (subs (u/slugify table-name) 0 (min (count table-name)
```

Comment:
there are just too many things we have to `deref` when reading this, can we break small things into meaningful `let` vars so it's easier to read?

things like `(t/format time-format (t/local-date-time))` could be called `date-time-postfix` would make it easier to read IMO.